### PR TITLE
add release_queue() to Client and Eventloop

### PR DESCRIPTION
**Pull request by andresv**
This PR adds `Client::release_queue()` and `Eventloop::release_queue()`.
Those can be called before dropping Client and Eventloop to get back original producer and consumer that can be given back to BBQueue after that.
Useful when you want to reinit everything from scratch.

```rust
let prod = mqtt_client.release_queue().unwrap();
let cons = mqtt_eventloop.release_queue().unwrap();
mqtt_queue.try_release_framed(prod, cons);

drop(mqtt_client);
drop(mqtt_eventloop);
```

### CODE REVIEW

File: mqttrust_core/src/client.rs [hunk 1]
```diff
@@ -23,14 +23,24 @@ use mqttrust::{
...
             client_id,
-            producer: RefCell::new(producer),
+            producer: Some(RefCell::new(producer)),
+        }
+    }
+
+    /// Release `FrameProducer`
+    ///
+    /// This can be called before dropping `Client` to get back original `FrameProducer`.
+    pub fn release_queue(&mut self) -> Option<FrameProducer<'a, L>> {
```
Comment (andresv) [inline mqttrust_core/src/client.rs:40]: Should it be `try_release_queue` because it returns option?

Comment (andresv) [reply mqttrust_core/src/client.rs:40]: No `try_` is used for something that returns `Result`.

**Review by MathiasKoch — APPROVED** [review]
LGTM :+1:

**Comment by andresv** [comment]
CI errors seems to be unrelated to my change.

**Comment by andresv** [comment]
Wanna merge?

**Pull request merged**